lib/tasks/spellcheck_task.rb

```diff
@@ -0,0 +1,170 @@
+#--
+# Yast rake
+#
+# Copyright (C) 2015 Novell, Inc.
+#   This library is free software; you can redistribute it and/or modify
+# it only under the terms of version 2.1 of the GNU Lesser General Public
+# License as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#++
+#
+# Rake task for checking spelling in the documentation files.
+# By default checks all *.md and *.html files.
+#
+# Supports custom dictionaries:
+#
+#  - global dictionary located in the tasks gem (lib/tasks/spell.yml)
+#  - repository specific dictionary (.spell.yml in the root directory)
+#
+# The custom dictionaries are defined at the "dictionary" key.
+#
+
+require 'rake'
+require 'rake/tasklib'
+
+module Yast
+  # Defines a spellcheck rake task
+  class SpellcheckTask < Rake::TaskLib
+    GLOBAL_SPELL_CONFIG_FILE = File.expand_path("../spell.yml", __FILE__)
+    CUSTOM_SPELL_CONFIG_FILE = ".spell.yml"
+
+    # define the Rake task in the constructor
+    def initialize
+      namespace :check do
+        desc "Run spell checker (by default for *.md and *.html files)"
+        task :spelling do
+          run_task
+        end
+      end
+
+    end
+
+    private
+
+    # optionally colorize the misspelled words if the rainbow gem is present
+    # @return [Boolean] true when the colorization support is present
+    def colorize
```

Comment:
from style guide question mark should be used for predicate methods. For me it is just implementation detail if colorize is attribute or somehow magically recognized by term capability.